### PR TITLE
modules/hal_st: Align sensor drivers to stmemsc HAL i/f v2.02

### PR DESCRIPTION
**Pull request by avisconti**
Align all sensor drivers that are using stmemsc (STdC) HAL i/f to new APIs of stmemsc v2.02.

Requires https://github.com/zephyrproject-rtos/hal_st/pull/13

### CODE REVIEW

File: tests/drivers/build_all/sensor/app.overlay [hunk 1]
```diff
@@ -146,71 +146,16 @@
  * pristine
  */
 
-/* disable device to conflict with i2c version */
```
Comment (erwango) [inline tests/drivers/build_all/sensor/app.overlay:149]: Can you explain why are these removed ?
Are they built elsewhere ?

Comment (avisconti) [reply tests/drivers/build_all/sensor/app.overlay:149]: > Can you explain why are these removed ? Are they built elsewhere ?

I remove the disabling of i2c, since I want the test to be done forboth busses.
I do not recall why originally the test excluded I2C for the sensors declared on SPI as well, but I feel that was because multi-istance was not working properly (or not at all) at that time. If you see, I still left a couple of STM sensors (ism330dhcx and iis2dh) with I2C disabled, as there are still issues when both i2c and spi are instanced. I will solve those issue as soon as possible, but not in this PR

Comment (erwango) [reply tests/drivers/build_all/sensor/app.overlay:149]: :-). My bad, commented to quickly. I didn't see that nodes were disabled initially.

**Review by erwango — DISMISSED** [review]

File: tests/drivers/build_all/sensor/app.overlay (hunk 1 above)
Comment (avisconti) [reply tests/drivers/build_all/sensor/app.overlay:149]: > :-). My bad, commented to quickly. I didn't see that nodes were disabled initially.

Thanks. I just re-pushed with no changes, but after rebasing on latest main branch.
(your approval disappeared... :D )

**Comment by avisconti** [comment]
@erwango can you please take a look at it again? I repushed it and your approval requires to be done again

**Comment by erwango** [comment]
@avisconti Would you mind rebasing once again ?

File: tests/drivers/build_all/sensor/app.overlay (hunk 1 above)
Comment (MaureenHelm) [reply tests/drivers/build_all/sensor/app.overlay:171]: > I do not recall why originally the test excluded I2C for the sensors declared on SPI as well, but I feel that was because multi-instance was not working properly (or not at all) at that time. But now I want to test both cases.

Can you mention this in the commit message please?

**Review by MaureenHelm — DISMISSED** [review]

File: tests/drivers/build_all/sensor/app.overlay (hunk 1 above)
Comment (MaureenHelm) [reply tests/drivers/build_all/sensor/app.overlay:171]: Thank you!

**Review by teburd — DISMISSED** [review]

**Comment by avisconti** [comment]
@MaureenHelm @teburd Thanks for approval. The PR is currently blocked by this one: https://github.com/zephyrproject-rtos/hal_st/pull/13. Can you possibly take a look at it? Thanks very much.

**Comment by erwango** [comment]
@avisconti https://github.com/zephyrproject-rtos/hal_st/pull/13 is now merged, please update west.yml

**Comment by avisconti** [comment]
> @avisconti [zephyrproject-rtos/hal_st#13](https://github.com/zephyrproject-rtos/hal_st/pull/13) is now merged, please update west.yml

Done, thx! I also removed the DNM

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
